src/main/java/com/yelp/nrtsearch/server/luceneserver/ShardState.java

```diff
@@ -939,6 +940,7 @@ public synchronized void startReplica(ReplicationServerClient primaryAddress, lo
           new HostPort(
               indexState.getGlobalState().getHostName(),
               indexState.getGlobalState().getReplicationPort());
+      LuceneServerConfiguration configuration = indexState.getGlobalState().getConfiguration();
```

Comment:
There are two more usages of this variable before. This can be defined before line 915 and and used in those places too.